test/iidm/TieLineTest.cpp

```diff
@@ -363,9 +369,65 @@ BOOST_AUTO_TEST_CASE(adder) {
     POWSYBL_ASSERT_THROW(tieLineAdder.add(), ValidationException, "AC tie line 'UNIQUE_TIE_LINE_ID': xnodeQ is not set for half line 2");
     tieLineAdder.setXnodeQ(40.0);
 
+    tieLineAdder.setFictitious(true);
+
     BOOST_CHECK_NO_THROW(tieLineAdder.add());
     POWSYBL_ASSERT_THROW(tieLineAdder.add(), PowsyblException, "The network test already contains an object 'TieLine' with the id 'UNIQUE_TIE_LINE_ID'");
     BOOST_CHECK_EQUAL(2UL, network.getLineCount());
+
+    TieLine& line = static_cast<TieLine&>(network.getLine("UNIQUE_TIE_LINE_ID"));
+    BOOST_CHECK(!line.getHalf(Branch::Side::ONE).isFictitious());
+    BOOST_CHECK(line.getHalf(Branch::Side::TWO).isFictitious());
+    BOOST_CHECK(!line.isFictitious());
+}
+
+BOOST_AUTO_TEST_CASE(fictitious) {
+    Network network = createTieLineTestNetwork();
+
+    TieLineAdder tieLineAdder = network.newTieLine();
+    tieLineAdder.setVoltageLevel1("VL2");
+    tieLineAdder.setVoltageLevel2("VL4");
+    tieLineAdder.setBus1("");
+    tieLineAdder.setConnectableBus1("VL2_BUS1");
+    tieLineAdder.setBus2("");
+    tieLineAdder.setConnectableBus2("VL4_BUS1");
+    tieLineAdder.setUcteXnodeCode("UcteXnodeCodeTest");
+    tieLineAdder.setName("TL_VL2_VL4");
+    tieLineAdder.setId("UNIQUE_TIE_LINE_ID");
+    tieLineAdder.setFictitious(true); // test that generated TieLine can be fictitious
+
+    //Half1
+    tieLineAdder.line1();
+    tieLineAdder.setName("H1_TL_VL2_VL4");
+    tieLineAdder.setId("H1_TL_VL2_VL4");
+    tieLineAdder.setB1(4.0);
+    tieLineAdder.setB2(5.0);
+    tieLineAdder.setG1(2.0);
+    tieLineAdder.setG2(3.0);
+    tieLineAdder.setR(60.0);
+    tieLineAdder.setX(660.0);
+    tieLineAdder.setXnodeP(10.0);
+    tieLineAdder.setXnodeQ(20.0);
+    tieLineAdder.setFictitious(true);
+
+    //Half2
+    tieLineAdder.line2();
+    tieLineAdder.setName("H2_TL_VL2_VL4");
+    tieLineAdder.setId("H2_TL_VL2_VL4");
+    tieLineAdder.setB1(7.0);
+    tieLineAdder.setB2(12.0);
+    tieLineAdder.setG1(6.0);
+    tieLineAdder.setG2(9.0);
+    tieLineAdder.setR(70.0);
+    tieLineAdder.setX(700.0);
+    tieLineAdder.setXnodeP(30.0);
+    tieLineAdder.setXnodeQ(40.0);
+    tieLineAdder.setFictitious(false);
+
+    TieLine& line = tieLineAdder.add();
+    BOOST_CHECK(line.getHalf(Branch::Side::ONE).isFictitious());
+    BOOST_CHECK(!line.getHalf(Branch::Side::TWO).isFictitious());
+    BOOST_CHECK(line.isFictitious());
```

Comment:
I don't know if it's voluntary, but you could use the fluent API.

Reply:
It wasn't voluntary => fixed